pkg/types/nutanix/validation/platform.go

```diff
@@ -0,0 +1,85 @@
+)
+
+// ValidatePlatform checks that the specified platform is valid.
+// TODO(nutanix): Revisit for further expanding the validation logic
+func ValidatePlatform(p *nutanix.Platform, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(p.PrismCentral) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismCentral"), "must specify the Prism Central"))
+	}
+	if len(p.PrismElement) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElement"), "must specify the Prism Element"))
+	}
+	if len(p.Username) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("username"), "must specify the username"))
+	}
+	if len(p.Password) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("password"), "must specify the password"))
+	}
+	if len(p.Port) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("port"), "must specify the port"))
+	}
+	if len(p.Subnet) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("subnet"), "must specify the subnet"))
+	}
+	if len(p.PrismCentral) != 0 {
+		if err := validate.Host(p.PrismCentral); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("prismCentral"), p.PrismCentral, "must be the domain name or IP address of the Prism Central"))
+		}
+	}
+
+	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
+	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
+		allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+	}
+
+	return allErrs
+}
+
+// ValidateForProvisioning checks that the specified platform is valid.
+func ValidateForProvisioning(p *nutanix.Platform, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(p.PrismElement) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("prismElements"), "must specify the cluster"))
+	}
+
+	if len(p.Subnet) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("subnet"), "must specify the subnet"))
+	}
+
+	allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+	return allErrs
+}
+
+// validateVIPs checks that all required VIPs are provided and are valid IP addresses.
+func validateVIPs(p *nutanix.Platform, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(p.APIVIP) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("apiVIP"), "must specify a VIP for the API"))
+	} else if err := validate.IP(p.APIVIP); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, err.Error()))
+	}
+
+	if len(p.IngressVIP) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("ingressVIP"), "must specify a VIP for Ingress"))
+	} else if err := validate.IP(p.IngressVIP); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("ingressVIP"), p.IngressVIP, err.Error()))
+	}
+
+	if len(p.APIVIP) != 0 && len(p.IngressVIP) != 0 && p.APIVIP == p.IngressVIP {
```

Comment:
At this point in the code, we are already guaranteed that at least one of the VIPs are non-empty.
```suggestion
	if p.APIVIP == p.IngressVIP {
```